core/drivers/crypto/ele/ele.c

```diff
@@ -495,33 +497,73 @@ static TEE_Result imx_ele_rng_get_trng_state(void)
 		return TEE_SUCCESS;
 }
 
-unsigned long plat_get_aslr_seed(void)
+/*
+ * Get random data from the EdgeLock Enclave.
+ *
+ * This function can be called when the MMU is off or on.
+ * virtual/physical address translation and cache maintenance
+ * is performed if needed.
+ *
+ * @buffer: data output
+ * @size: RNG data size
+ */
+static TEE_Result imx_ele_rng_get_random(uint8_t *buffer, size_t size)
 {
 	TEE_Result res = TEE_ERROR_GENERIC;
-	uint64_t timeout = timeout_init_us(10 * 1000);
+	struct imx_ele_buf rng = {};
+
 	struct rng_get_random_cmd {
 		uint32_t addr_msb;
 		uint32_t addr_lsb;
 		uint32_t size;
-		uint32_t crc;
 	} cmd = { };
 	struct imx_mu_msg msg = {
 		.header.version = ELE_VERSION_HSM,
 		.header.size = SIZE_MSG_32(cmd),
 		.header.tag = ELE_REQUEST_TAG,
 		.header.command = ELE_CMD_RNG_GET,
 	};
-	unsigned long aslr __aligned(CACHELINE_SIZE) = 0;
 
-	/*
-	 * This function can only be called when the MMU is off. No
-	 * virtual/physical address translation is performed, nor cache
-	 * maintenance.
-	 */
-	assert(!cpu_mmu_enabled());
+	if (!buffer || !size)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (cpu_mmu_enabled()) {
+		res = imx_ele_buf_alloc(&rng, NULL, size);
+		if (res != TEE_SUCCESS)
+			return res;
+
+		cmd.addr_msb = rng.paddr_msb;
+		cmd.addr_lsb = rng.paddr_lsb;
+	} else {
+		paddr_t pa = (paddr_t)buffer;
+
+		if (!IS_ALIGNED_WITH_TYPE(pa, uint32_t))
+			return TEE_ERROR_BAD_PARAMETERS;
+
+		reg_pair_from_64((uint64_t)pa, &cmd.addr_msb, &cmd.addr_lsb);
+	}
 
-	reg_pair_from_64((uint64_t)&aslr, &cmd.addr_msb, &cmd.addr_lsb);
-	cmd.size = sizeof(aslr);
+	cmd.size = (uint32_t)size;
+
+	memcpy(msg.data.u8, &cmd, sizeof(cmd));
+
+	res = imx_ele_call(&msg);
+	if (res)
+		goto out;
+
+	if (cpu_mmu_enabled())
+		res = imx_ele_buf_copy(&rng, buffer, size);
+out:
+	imx_ele_buf_free(&rng);
+
+	return res;
+}
+
+unsigned long plat_get_aslr_seed(void)
+{
+	uint64_t timeout = timeout_init_us(10 * 1000);
+
```

Comment:
Please remove this empty line.

Reply:
Done.

Reply:
CI is reporting an error without new line: https://github.com/OP-TEE/optee_os/actions/runs/13407450824/job/37449890100?pr=7268#step:5:61

Reply:
@jenswi-linaro please help me with this.

Reply:
The checkpatch warning looks like a false positive.
I usually run checkpatch as `./scripts/checkpatch.sh github/master..`, if `github/master` is the base of my patches.

Reply:
@jenswi-linaro IIUC this error is coming from `checkdiff` script.
`checkpatch` is passing correctly.

Reply:
I guess you can source `scripts/checkpatch_inc.sh` in bash and call the function from the prompt.

Reply:
Thanks @jenswi-linaro. I am able to run the `checkdiff` on my local machine.
It is also reporting the same issue:
```
:~/optee-os$ checkdiff acdc32afe18f8b71915184a01e7279ffcbe0ec20 47d18800fe372fdb4fad575daba61262e814d171
WARNING: Missing a blank line after declarations
#669: FILE: core/drivers/crypto/ele/ele.c:564:
+       uint64_t timeout = timeout_init_us(10 * 1000);
+       unsigned long aslr __aligned(CACHELINE_SIZE) = 0;

total: 0 errors, 1 warnings, 0 checks, 932 lines checked
:~/optee-os$
```

So, we should ignore it, right ?

Reply:
Yes

Reply:
Ok. Thanks @jenswi-linaro